service/storage_service.cc

```diff
@@ -714,14 +714,14 @@ class topology_coordinator {
     future<> cleanup_group0_config_if_needed() {
         auto& topo = _topo_sm._topology;
         auto rconf = _group0.group0_server().get_configuration();
-        if (rconf.current.size() > topo.normal_nodes.size() + topo.new_nodes.size() + topo.transition_nodes.size()) {
-            // Raft config is larger than the sum of all nodes not in 'left' state.
+        auto to_remove = boost::copy_range<std::unordered_set<raft::server_id>>(
+                boost::range::join(rconf.current, rconf.previous)
```

Comment:
Why do we need to check `previous`? If previous is not empty there is a config transition. We should do nothing.